zio-cli/shared/src/main/scala/zio/cli/FigFontParser.scala

```diff
@@ -0,0 +1,336 @@
+package zio.cli
+
+import scala.collection.mutable
+
+// FigLettering Font (.flf), see:
+// http://www.jave.de/figlet/figfont.html or
+// https://github.com/cmatsuoka/figlet/blob/master/figfont.txt
+final case class FigFont(
+  header: FigHeader,
+  rightToLeft: Boolean,
+  layout: Layout,
+  chars: Map[Char, FigChar]
+)
+
+final case class FigChar(lines: Seq[FigCharLine], width: Int, height: Int)
+
+final case class FigCharLine(leftSpaces: Int, rightSpaces: Int, content: String)
+
+case class FigHeader(
+  signature: String,
+  hardBlank: Char,
+  charHeight: Int,
+  baseline: Int,
+  maxLength: Int,
+  commentLines: Int,
+  oldLayoutMask: Int,
+  rightToLeft: Option[Int],
+  fullLayoutMask: Option[Int]
+)
+
+final case class Layout(horizontal: HorizontalLayout, verticalLayout: VerticalLayout)
+
+final case class VerticalLayout(
+  mode: LayoutMode,
+  equalCharSmushing: Boolean,
+  underscoreSmushing: Boolean,
+  heirarchySmushing: Boolean,
+  horizontalLineSmushing: Boolean,
+  verticalLineSuperSmushing: Boolean,
+  universalSmushing: Boolean
+)
+
+final case class HorizontalLayout(
+  mode: LayoutMode,
+  equalCharSmushing: Boolean,
+  underscoreSmushing: Boolean,
+  heirarchySmushing: Boolean,
+  oppositePairSmushing: Boolean,
+  bigXSmushing: Boolean,
+  hardBlankSmushing: Boolean,
+  universalSmushing: Boolean
+)
+
+sealed trait LayoutMode
+
+object LayoutMode {
+  object FullSize extends LayoutMode
+  object Fitted   extends LayoutMode
+  object Smushed  extends LayoutMode
+}
+
+object FigFontParser {
+  import ParseLib._
+
+  def parse(lines: Seq[String]): Either[String, FigFont] =
```

Comment:
Maybe we could move this to `FigFont`, as a constructor:

```scala
object FigFont {
  ...
  def fromFile(content: String): Either[String, FigFont] = ???
  def fromLines(content: Iterable[String]): Either[String, FigFont] = ???
  ...
}
```

Reply:
- [ ] add `object FigFont` with publish API